### DISCUSSION TITLE
import untagged_vlan and tagged_vlans in csv for interface

### DISCUSSION THREAD

**Discussion by vaas2956** [netbox-community/netbox#13198] [Q&A]
Hello team, 

I've been working on Netbox for a while, mainly on the export part.

Recently, we have received new informations regarding tagged_vlans and tagged_vlan for interfaces. I saw that we could make mass changes on csv interfaces. However, I can't change the tagged_vlans and untagged_vlan but I can change the mode, the enabled etc ... 

Example of a query I would like to make :
name,device,type,untagged_vlan
13,"TAP",1000base-t,150

Do you know how to import this in csv for interface ? 

Best regards

**Comment by candlerb** [comment]
What exact error do you see? Or does it silently do nothing? I'm guessing the latter.

If you go to `/dcim/interfaces/import/` you'll see that `untagged_vlan` is not an available column. However, there's currently no error generated if you try to upload a non-existent column: ~~I've just raised this as #13200~~ see #11617 (although the issue name is not very descriptive).

As to why you can import "mode" but not "untagged_vlan" or "tagged_vlans": I see that `class InterfaceImportForm` does not include them. Adding them would have to be a feature request. Otherwise, you may be able to do it via the REST API.

**Reply by vaas2956** [reply]
Indeed I misexpressed the problem sorry my English is not great. 
So I have to go through the REST API for my request.
thank you for your help

**Reply by candlerb** [reply]
Or you could write a [custom script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) which accepts a [file upload](https://docs.netbox.dev/en/stable/customization/custom-scripts/#filevar); in that case you'd be working with the Django ORM rather than the REST API. This means that end-users can access the feature directly through the Netbox web UI, and also that you can do all your work within a database transaction.
